aiven_mysql_migrate/migration.py

```diff
@@ -180,11 +180,15 @@ def _check_bin_log_format(self):
             if row_format.upper() != "ROW":
                 raise UnsupportedBinLogFormatException(f"Unsupported binary log format: {row_format}, only ROW is supported")
 
-    def run_checks(self) -> MySQLMigrateMethod:
-        """Raises an exception if one of the the pre-checks fails, otherwise a method to be used for migration"""
+    def run_checks(self, fallback_to_dump_method: Optional[bool] = True) -> MySQLMigrateMethod:
+        """Raises an exception if one of the the pre-checks fails, otherwise a method to be used for migration.
+        If fallback_to_dump_method is False, re-raises validation exceptions in case if replication is not possible"""
```

Comment:
Nit: one empty line between title and description. Oh well, one can argue it's all description.